joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -0,0 +1,301 @@
+      val argumentAsts     = argumentLiterals.map(astForExpression)
+      val call =
+        callNode(node, node.text, Operators.arrayInitializer, Operators.arrayInitializer, DispatchTypes.STATIC_DISPATCH)
+      callAst(call, argumentAsts)
+    }
+  }
+
+  protected def astForHashLiteral(node: HashLiteral): Ast = {
+    val argumentAsts = node.elements.flatMap(ctx =>
+      ParserAst(ctx) match
+        case associationNode: Association => astForAssociation(associationNode) :: Nil
+        case node =>
+          logger.warn(s"Could not represent element: ${node.text} ($relativeFileName), skipping")
+          astForUnknown(node) :: Nil
+    )
+    val call = callNode(
+      node,
+      node.text,
+      RubyOperators.hashInitializer,
+      RubyOperators.hashInitializer,
+      DispatchTypes.STATIC_DISPATCH
+    )
+    callAst(call, argumentAsts)
+  }
+
+  protected def astForAssociation(node: Association): Ast = {
+    val key   = astForExpression(node.key)
+    val value = astForExpression(node.value)
+    val call =
+      callNode(node, node.text, RubyOperators.association, RubyOperators.association, DispatchTypes.STATIC_DISPATCH)
+    callAst(call, Seq(key, value))
+  }
+
+  protected def astForUnknown(node: ParserNode): Ast = {
+    val className = node.getClass.getSimpleName
+    val text      = node.text
+    logger.warn(s"Could not represent expression: $text ($className) ($relativeFileName), skipping")
+    Ast(unknownNode(node, text))
+  }
+
+  private def astForMemberCallWithoutBlock(node: SimpleCall, memberAccess: MemberAccess): Ast = {
+    val receiverAst    = astForFieldAccess(memberAccess)
+    val methodName     = memberAccess.methodName
+    val methodFullName = methodName // TODO
+    val argumentAsts   = node.arguments.map(astForExpression)
+    val call           = callNode(node, node.text, methodName, methodFullName, DispatchTypes.STATIC_DISPATCH)
+    callAst(call, argumentAsts, None, Some(receiverAst))
+  }
+
+  private def astForMethodCallWithoutBlock(node: SimpleCall, methodIdentifier: SimpleIdentifier): Ast = {
+    val methodName     = methodIdentifier.ctx.getText
+    val methodFullName = methodName // TODO
+    val argumentAst    = node.arguments.map(astForExpression)
+    val call           = callNode(node, node.text, methodName, methodFullName, DispatchTypes.STATIC_DISPATCH)
+    callAst(call, argumentAst, None, None)
+  }
+
+  protected def astForFieldAccess(node: MemberAccess): Ast = {
+    val fieldIdentifierAst = Ast(fieldIdentifierNode(node, node.methodName, node.methodName))
+    val targetAst          = astForExpression(node.target)
+    val code               = s"${node.target.getText}${node.op}${node.methodName}"
+    val fieldAccess = callNode(node, code, Operators.fieldAccess, Operators.fieldAccess, DispatchTypes.STATIC_DISPATCH)
+    callAst(fieldAccess, Seq(targetAst, fieldIdentifierAst))
+  }
+
+  private def getBinaryOperatorName(op: String): Option[String] = {
+    Map(
```

Comment:
This Map could be defined statically somewhere instead of creating it here with every call to `getBinaryOperatorName`.